view/frontend/web/js/model/shipping-save-processor/default.js

```diff
@@ -0,0 +1,73 @@
+
+define([
+    'jquery',
+    'ko',
+    'Magento_Checkout/js/model/quote',
+    'Magento_Checkout/js/model/resource-url-manager',
+    'mage/storage',
+    'Magento_Checkout/js/model/payment-service',
+    'Magento_Checkout/js/model/payment/method-converter',
+    'Magento_Checkout/js/model/error-processor',
+    'Magento_Checkout/js/model/full-screen-loader',
+    'Magento_Checkout/js/action/select-billing-address'
+], function(
+    $,
+    ko,
+    quote,
+    resourceUrlManager,
+    storage,
+    paymentService,
+    methodConverter,
+    errorProcessor,
+    fullScreenLoader,
+    selectBillingAddressAction
+) {
+    'use strict';
+
+    return {
+        saveShippingInformation: function() {
+            var payload;
+            var kl_sms_consent = $('[name="custom_attributes[kl_sms_consent]"]').is(':checked');
+            var kl_email_consent = $('[name="custom_attributes[kl_email_consent]"]').is(':checked');
+
+            if(!quote.billingAddress()) {
+                selectBillingAddressAction(quote.shippingAddress());
+            }
+
+            payload = {
+                addressInformation: {
+                    shipping_address: quote.shippingAddress(),
+                    billing_address: quote.billingAddress(),
+                    shipping_method_code: quote.shippingMethod().method_code,
+                    shipping_carrier_code: quote.shippingMethod().carrier_code,
+                    extension_attributes: {
+                        kl_sms_consent: kl_sms_consent,
+                        kl_email_consent: kl_email_consent,
+                    }
+                }
+            };
+
+            //TODO Remove
+            console.log('debug sms', kl_sms_consent);
+            console.log('debug email', kl_email_consent);
+
+            fullScreenLoader.startLoader();
+
+            return storage.post(
+                resourceUrlManager.getUrlForSetShippingInformation(quote),
+                JSON.stringify(payload)
+            ).done(
+                function(response) {
+                    quote.setTotals(response.totals);
```

Comment:
why are we doing all the stuff here?  Is this replacing the existing functionality?  what if the customer has other extensions that run affecting what happens here i.e if they had a 3d party coupon processor

Reply:
yea this is basically magento standard for doing this based off all documentation and what similar apps do.